federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -187,3 +193,14 @@ func StartControllers(s *options.CMServer, restClientCfg *restclient.Config) err
 
 	select {}
 }
+
+// TODO(madhusudancs): Remove this in 1.5. This is only temporary to give an
+// upgrade path in 1.4.
+func restClientConfigFromSecret(master string) (*restclient.Config, error) {
+	kubeconfigGetter := util.KubeconfigGetterForSecret(DeprecatedKubeconfigSecretName)
+	restClientCfg, err := clientcmd.BuildConfigFromKubeconfigGetter(master, kubeconfigGetter)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find the Federation API server kubeconfig, tried the flags and the deprecated secret %s: %v", DeprecatedKubeconfigSecretName, err)
```

Comment:
nit: mention which flags? So that user knows which flag to look at amongst so many other flags :)